microsoft-edge/devtools-guide-chromium/whats-new/2023/01/devtools-109.md

```diff
@@ -0,0 +1,201 @@
+
+   ![Exporting a trace](./devtools-109-images/enhanced-trace-export.png)
+
+1. In the **Save As** dialog, save the new `.devtools` file.
+
+   Or, in the **Memory** tool, click the **Save** link next to the heap snapshot.  In the **Save As** dialog, save the new `.devtools` file.
+
+**Importing:**
+
+1. Import the `.devtools` file from within the **Performance** tool or **Memory** tool.  If you use the **Performance** tool, click the **Load profile** (![Load profile icon](./devtools-109-images/load-profile-icon.png)) button:
+
+   ![Importing a trace from within the Performance tool](./devtools-109-images/enhanced-trace-import.png)
+
+   If you use the **Memory** tool, right-click the **Heap Snapshots** pane and then select **Load**.
+
+1. A new DevTools window opens, containing a subset of the tools, including the **Performance** or **Memory** tool.  The **Elements**, **Console**, and **Sources** tools are pre-populated with the preserved state:
+
+   ![Trace import console](./devtools-109-images/enhanced-trace-import-console.png)
+
+This is a new feature that the DevTools team is actively experimenting with, so please [send us your feedback in our GitHub repo](https://github.com/MicrosoftEdge/DevTools/issues)!
+
+See also:
+* [Share performance and memory traces (PR 2388 draft)](https://github.com/MicrosoftDocs/edge-developer/blob/user/pabrosse/enhanced-traces/microsoft-edge/devtools-guide-chromium/evaluate-performance/share-traces.md)
+<!-- todo: link to final article when PR is live, after merge main into present branch.  expect:
+* [Share performance and memory traces](../../../evaluate-performance/share-traces.md)
+-->
+
+
+<!-- ====================================================================== -->
+## Debug long-running Recalculate Styles events in the Performance tool with new selector stats
+
+<!-- Subtitle: Understand which of your CSS selectors are contributing to slow performance of your website or app. -->
+
+In Microsoft Edge 109, in the **Performance** tool, you can turn on the **Enable advanced rendering instrumentation (slow)** setting in the **Performance** tool to access the new selector stats feature.
+
+With this setting turned on:
+
+1. Select **Record** and execute the scenario you want to improve on your website or app.
+
+1. Select **Stop**.
+
+1. Select a **Recalculate Style** event.  In the bottom section of the **Performance** tool, select the **Selector Stats** tab:
+
+   ![Performance tool Settings 'Enable advanced rendering instrumentation (slow)' checkbox, and 'Selector Stats' tab](./devtools-109-images/advanced-rendering-instrum.png)
+
+   The **Selector Stats** tab provides a list of all the CSS selectors that were calculated by the browser engine during the **Recalculate Style** event.  You can sort selectors by the **Elapsed** time they took to process, or by the number of elements they matched (the **Match Count** column).  Use this data to:
```

Comment:
```suggestion
The **Selector Stats** tab provides a list of all the CSS selectors that were calculated by the browser engine during the **Recalculate Style** event.  You can sort selectors by the **Elapsed** time they took to process, or by the number of elements they matched (the **Match Count** column).  Use this data to:
```
Feel like this was weirdly indented in the preview. Here's what I'm seeing: 
![image](https://user-images.githubusercontent.com/34316555/212152834-cd235b43-ad5d-448a-87f7-487fc0599987.png)

Reply:
changed